app/controllers/namespaces_controller.rb

```diff
@@ -11,20 +11,24 @@ class NamespacesController < ApplicationController
   # GET /namespaces
   # GET /namespaces.json
   def index
-    respond_to do |format|
-      format.html { skip_policy_scope }
-      format.json do
-        @special_namespaces = Namespace.where(
-          "global = ? OR namespaces.name = ?", true, current_user.username
-        ).order("created_at ASC")
-        @namespaces = policy_scope(Namespace).order(created_at: :asc)
-
-        accessible_json = serialize_as_json(@namespaces)
-        special_json = serialize_as_json(@special_namespaces)
-        render json: {
-          accessible: accessible_json,
-          special:    special_json
-        }
+    if request.head?
+      check_namespace_by_name if params[:name]
```

Comment:
@mssola You probably have something to comment regarding this.

Reply:
Yeah, this is not the way to go... Moreover, consider what I just said on the above comment.

I'd create a new endpoint that checks:

1. The availability of the name.
2. The name has a proper format.

This in turn could be extended to other resources (e.g. teams). This can be done with regular Rails for now, but maybe it would be a good starting point for grape on namespaces.